include/stdx/isnan.hpp

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Created by Tomas Zeman on 2/3/2018.
+ *  Copyright 2018 Two Blue Cubes Ltd. All rights reserved.
+ *
+ *  Distributed under the Boost Software License, Version 1.0. (See accompanying
+ *  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+ */
+#ifndef STDX_ISNAN_HPP_INCLUDED
+#define STDX_ISNAN_HPP_INCLUDED
+
+#ifdef __DJGPP__
+
+#include <cstdint>
+
+namespace stdx {
+
+    inline bool isnan(float f) {
+        static_assert(sizeof(float) == sizeof(uint32_t), "Important stdx::isnan(float) assumption violated");
+
+        uint32_t& n = *reinterpret_cast<uint32_t*>(&f);
```

Comment:
My intention was to give the compiler a better change to compile out the logic but I din't realize how smart compilers are these days. I can now confirm that it can compile this out even when using `std::memcpy` (at least gcc can). I've proven that using https://godbolt.org/.

Reply:
Even if it wasn't as fast, it is better to have correct code than fast code, but yeah -- GCC and Clang have seen through this for a long time, and often even generate better code (reg to reg move versus mem load).